### ISSUE TITLE
Crash on startup with Mystcraft

### ISSUE THREAD

**Issue by stevestech** [gamerforEA/Cauldron-Reloaded#14]
Cauldron: 1.1388.1.0-Fix1
Mystcraft: 1.7.10-0.11.5.00

This crash occurs on server startup when Mystcraft is installed.

Crash-log:
http://paste.ubuntu.com/10946611/

There has been some discussion about this on the Mystcraft github:
https://github.com/XCompWiz/Mystcraft-Issues/issues/66

**Comment by spannerman79** [comment]
AS Cauldron uses code from Spigot - as Cauldron has the ability to use world management via plugins - yes of course there will be issues while using Mystcraft.

The code from spigot has its own chunk management system in place which conflicts with Mystcraft.

This is part and parcel in using Cauldron - not everything Forge based will work out of the box on Cauldron.  

A side note though - if mystcraft is doing this now it will be an issue once Sponge is no longer a dev build.  And yes technically Cauldron is dead but the devs involved with Cauldron and Spigot are involved heavily with Sponge

This is the same sort of issue that was had with Spigot and some standard plugins - some worked some didn't.  Obviously Mystcraft doesn't work.

**Comment by stevestech** [comment]
Cauldron works fine with older versions of Mystcraft, and modded dimensions in general. The problem with build 0.11.5.00 is that it adds a profiling dimension called mystdummy. This dimension generates empty chunks. According to the following console errors / crash report, it is the empty chunks which Cauldron does not like.

http://paste.ubuntu.com/10955203/
http://paste.ubuntu.com/10955457/

As for the missing field crash, this should be pretty easy to diagnose once that field is identified surely?
http://paste.ubuntu.com/10946611/

**Comment by spannerman79** [comment]
I *think* that the field was removed so that world management plugins could handle chunk loading on dummy or custom dimensions/worlds on 1.7.10.  I don't know 100% of course and Blood ( http://cauldron.minecraftforge.net/threads/downloads-removed-working-on-sponge-1-8.780/ ) is busy with Sponge and his/her focus is on getting Sponge out - for which there is a dev alpha release already ( https://forums.spongepowered.org/t/sponge-dev-builds-now-available/6811 )

The plugins i mention are Multiverse and Multiworld of course

**Comment by jburrows82** [comment]
Regardless of reasoning, this is on Mystcraft issue as shown on their tracker. If you want compatibility with Cauldron and other mods, revert. If you want performance from Mystcraft, use vanilla forge. This issue should be closed as the fault lies on Mystcraft's side

**Comment by spannerman79** [comment]
Exactly @jburrows82 .  Even @Shybella showed the incompatibilities with Mystcraft with other mods and yet the mod dev doesn't want to do anything about it sighting "show logs" instead of checking compatibility herself/himself before creating a custom chunk manager

The mod dev should take notes from Chickenbones when it comes to chunk managment

Anyway I am digressing and this issue should be closed.

As already stated by @jburrows82 - if you want full 100% compatibility with forge mods - use vanilla forge.  Or learn to compromise (until Sponge is ready)

**Comment by Myrninvollo** [comment]
@gamerforEA @stevestech please close this issue, proven its Mystcraft. thx

**Comment by Shybella** [comment]
Yeah, xCompWiz's ignorance is something else. Glad the issue was resolved.

**Comment by dezil** [comment]
there is a work around for this you just need to run the server with vanilla forge for the first 20 mins or so while it gens the mystcraft baseline world, can then run with cauldron fine

**Comment by Prototik** [comment]
[I Just Leave Some Link Here](https://prok.pw/cauldron-1.7.10-2.1403.1.49-installer.jar)

**Comment by spannerman79** [comment]
@Prototik where did the source for that come from?

**Comment by Prototik** [comment]
@spannerman79 It's my own build with mystcraft related fixes (and some others).

**Comment by spannerman79** [comment]
For the sake of trust - do you have a Github or location showing your source changes @Prototik ?

**Comment by Prototik** [comment]
@spannerman79 Not yet, but soon. You could view my profile - i'm not person who will fool you :)

btw my build has one great fixes: almost all version of mystcraft on cauldron has a bug  with redstone and mechanisms in mystcraft's dimensions (not working at all), my build has fixes for it.

**Comment by spannerman79** [comment]
Don't take it the wrong way but with how some people are - even more so now that Blood isn't behind Cauldron and working on Sponge - the number of "dodgy" builds have increased.

Just like this Github where the source can be viewed I trust these builds, as anyone can go through and point out anything that might be amiss code wise.

**Comment by Prototik** [comment]
@spannerman79 just *I* don't trust github with some... hmm... illegal code. I'll publish my sources on own github-like service soon, but for now I can only provide final "doggy" builds. You can choose - use my build, use outdated mystcraft with bugs or throw mystcraft from server at all. I do not force you to use my build.

**Comment by spannerman79** [comment]
Thats fine re github & source

Perhaps something got lost in the translation when I mentioned the direction Blood has gone so I'll leave it at that.

**Comment by Shybella** [comment]
I'm interested too

**Comment by Yive** [comment]
@Prototik Is there a way for me to contact you privately? If you don't want to post your email or whatnot, my twitter is ItsYive so that you can DM me there.

**Comment by Prototik** [comment]
@Yive my email is on github profile: me@prok.pw

**Comment by Prototik** [comment]
@spannerman79 @Shybella [I've only just opened the source](https://prok.pw/git/Prototik/KCauldron)

**Comment by Cracks777** [comment]
@Prototik  where are the downloads?

**Comment by Cracks777** [comment]
Prototiks version is forked from this one

**Comment by TheBrain0110** [comment]
@Prototik Since it's a fork, any possibility of making a pull request back to this one so we can have your patches in the main line?

**Comment by Prototik** [comment]
@TheBrain0110 No, I'll support my own version.

**Comment by Prototik** [comment]
@Cracks777 downloads under "Releases" tab.

**Comment by Cracks777** [comment]
ty, i see, it took some of them a while to show up.

**Comment by Cracks777** [comment]
@Prototik does your version have the same patches as this one plus some?

**Comment by Prototik** [comment]
@Cracks777 yes.
